common/lib/capa/capa/xqueue_interface.py

```diff
@@ -134,12 +136,17 @@ def _send_to_queue(self, header, body, files_to_upload):
 
     def _http_post(self, url, data, files=None):
         try:
-            r = self.session.post(url, data=data, files=files)
+            response = self.session.post(url, data=data, files=files,
+                                         timeout=(CONNECT_TIMEOUT, READ_TIMEOUT))
         except requests.exceptions.ConnectionError, err:
             log.error(err)
             return (1, 'cannot connect to server')
 
-        if r.status_code not in [200]:
+        except requests.exceptions.ReadTimeout, err:
+            log.error(err)
+            return (1, 'Failed to read from the server')
+
+        if response.status_code not in [200]:
             return (1, 'unexpected HTTP status code [%d]' % r.status_code)
```

Comment:
If you're changing `r` to `response`, you need to change the reference to `r.status_code` on the `return` line as well.